### ISSUE TITLE
Figure out what's happening with preprocessing memory usage.

### ISSUE THREAD

**Issue by KerekesDavid** [yurujaja/pangaea-bench#22]
That's an insane amount of memory to be saved on such a tiny thing, good catch!

Just a quick napkin calculation, a 1024px, 4(?) channel image stored in double should take 1024*1024*12*64 / 8 / 1000^2 = ~33MB of memory. If you are tiling these to 224x224 patches, that's 25 images per big image, so that's 825MB per worker. with 8 workers, that comes up to 6GB of memory, which is about 1/4 of what you see here. 

It's a good catch, but I would like to make sure I understand what's going on here. The pattern of memory usage going _down_ during training is also very interesting. Python manages memory by ref-counting, so it can't be really blamed on garbage collection or anything.  

When I implemented I checked if pytorch tries to keep computation graph info about these tensors, and `requires_grad` came back as `False`. Not sure what I want to check next, but it seems like there are some issues to be discovered here. I wonder what using .detach(), or requires_grad=False wrapper around the processing pipeline would do to the memory usage.

_Originally posted by @KerekesDavid in https://github.com/yurujaja/geofm-bench/issues/21#issuecomment-2331769567_
